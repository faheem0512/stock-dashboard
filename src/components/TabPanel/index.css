.tab-container {
  display: flex;
  height: 60px;
  flex: 1;
}
.tab {
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  color: #ffffff;
}

.tab-panel {
  display: flex;
  background-color: #b3b3b3;
  font-size: 1.25rem;
}
