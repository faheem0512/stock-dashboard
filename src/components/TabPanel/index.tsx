import React from 'react';
import { NavLink } from 'react-router-dom';
import './index.css';

export interface ITabPanel {
  testId?: string;
  children: any;
}
export interface ITab {
  route: string;
  name: string;
}

export const Tab: React.FC<ITab> = ({ route, name }) => {
  return (
    <NavLink
      to={route}
      className="tab"
      activeStyle={{
        fontWeight: 'bold',
        backgroundColor: '#797979',
      }}
    >
      {name}
    </NavLink>
  );
};

const TabPanel: React.FC<ITabPanel> = ({ children, testId }) => {
  return (
    <div className="tab-panel" data-testid={testId}>
      <div className="tab-container">
        {React.Children.map(children, (child) => {
          const { props, type } = child;
          if (type === Tab) {
            return <Tab {...props} />;
          }
          return null;
        })}
      </div>
    </div>
  );
};

export default TabPanel;
