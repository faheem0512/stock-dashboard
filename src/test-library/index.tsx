import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import React from 'react';

export * from '@testing-library/react';

const Router = ({ children }: { children: JSX.Element }) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};

export const renderWithRouter = (ui: any, options?: any) => {
  return render(ui, { wrapper: Router, ...options });
};
