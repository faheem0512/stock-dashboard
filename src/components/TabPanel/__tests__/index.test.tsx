import React from 'react';
import { renderWithRouter } from 'test-library';
import TabPanel, { Tab } from '../index';

describe('Tab Panel', () => {
  it('tab panel render without error', () => {
    const { container } = renderWithRouter(
      <TabPanel testId="tabPanel">
        <Tab name="One" route="/one" />
        <Tab name="Two" route="/two" />
      </TabPanel>
    );
    expect(container).toMatchSnapshot();
  });
});
