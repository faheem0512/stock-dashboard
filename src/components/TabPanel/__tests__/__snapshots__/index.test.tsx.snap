// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Panel tab panel render without error 1`] = `
<div>
  <div
    class="tab-panel"
    data-testid="tabPanel"
  >
    <div
      class="tab-container"
    >
      <a
        class="tab"
        href="/one"
      >
        One
      </a>
      <a
        class="tab"
        href="/two"
      >
        Two
      </a>
    </div>
  </div>
</div>
`;
